### PR TITLE
feat(helm): add --skip-schema-validation flag to helm 'install', 'uprade' and 'lint'

### LINKED ISSUE #10398: Proposal: disable values schema validation for charts including a values.schema.json file
Even when a chart designer/maintainer has decided to include a ``values.schema.json`` file in the Helm chart he/she releases it could be that the user of that chart (using it directly or as a dependency in an integration/umbrella Helm chart) cannot assure to fulfill the defined schema for the deployment specific values passed to that subchart (e.g. he/she passes a deployment specific values.yaml file to the ``helm install/upgrade`` command applied on that chart, where some "old" Helm parameter removed from the ``values.schema.json`` of a newer version of that (sub)chart still remains).

I think it would be useful to support a new option in ``helm install``, ``helm upgrade``, ``helm lint`` and ``helm template`` allowing to disable the values schema validation (in case a ``values.schema.json`` file is packaged in the target Helm chart or some listed subchart)

### PR DESCRIPTION
**Pull request by anessi**
**What this PR does / why we need it**:

When --skip-schema-validation is set, any schema contained in the helm chart is ignored. Defaults to 'false'.

Closes #10398

**Special notes for your reviewer**:

I tried to keep the code in the `pkg` folder backwards compatible.

Supersedes https://github.com/helm/helm/pull/11510

Documentation added in https://github.com/helm/helm-www/pull/1549

**If applicable**:
- [x] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

----
### maintainer addition
- closes #11510

### CODE REVIEW

**Review by tleed5 — COMMENTED** [review]
I'm not a maintainer of Helm but I'm reviewing PRs to get more knowledge in the codebase and to help out a little bit. 

I didn't spot any obvious issues with the code changes and nothing came up while I was testing locally with the CLI and SDK

I think I prefer your approach over https://github.com/helm/helm/pull/11510 since it's more cleaner and handles backwards compatibility a bit better

**Comment by anessi** [comment]
@tleed5: thanks for your feedback.

For me it's not clear how to get [attention on PRs from a maintainer.](https://github.com/helm/helm/blob/main/CONTRIBUTING.md#pr-lifecycle) Looking at the growing number of open PRs it seems they are piling up. Is it the number of votes on the PR maybe? I could ask on the open issue for more votes as it already has 42 votes where on this PR we only have 8. Thanks for your input on this.

**Comment by tleed5** [comment]
@anessi I'm fairly new to this space too but from what I've seen posting a message in #helm-dev slack channel is a way of getting more attention on your PR

**Review by sabre1041 — APPROVED** [review]
LGTM

**Comment by joebowbeer** [comment]
@anessi can you clarify in the title if this also applies to `helm template`?

Note that `helm template` _is_ included in the #10398 description -- and it is what I need 😄

**Review by joebowbeer — COMMENTED** [review]
@anessi can you clarify in the title if this also applies to `helm template`?

Note that `helm template` is included in the https://github.com/helm/helm/issues/10398 description -- and it is what I need 😄

**Comment by anessi** [comment]
@joebowbeer `template` includes all command line args from `install`, so this is covered as well. See https://github.com/helm/helm/blob/main/cmd/helm/template.go#L193

I also tested it with an additional unit test. I can add it if you like.

**Review by joebowbeer — APPROVED** [review]
LGTM

**Review by onedr0p — APPROVED** [review]
LGTM

**Comment by anessi** [comment]
@sabre1041 @joebowbeer  As I don't have write permissions, can your merge this please - or is there anything else still pending?
It goes together with https://github.com/helm/helm-www/pull/1549 so this should be merged as well.

Thanks!

**Comment by sabre1041** [comment]
> @sabre1041 @joebowbeer As I don't have write permissions, can your merge this please - or is there anything else still pending? It goes together with [helm/helm-www#1549](https://github.com/helm/helm-www/pull/1549) so this should be merged as well.
> 
> Thanks!

An additional maintainer will need to complete a review of the PR before it can be integrated. I'll see who I can get to review

**Review by vepatel — APPROVED** [review]

**Comment by anessi** [comment]
@sabre1041 : did you find someone to look at the PR? thx

**Comment by sabre1041** [comment]
> @sabre1041 : did you find someone to look at the PR? thx

Thanks for the ping. Taking a look this weekend

**Review by scottrigby — APPROVED** [review]
@anessi thanks for building on the work in #11510. 

This PR looks good: solves a real need, takes a backwards-compatible approach (thank you), and passes all tests 👍

It may be useful to add an air-gapped section to the Helm docs as a follow-up.

**Comment by scottrigby** [comment]
added gh close keyword for #11510 to this PR description and noted as such

**Comment by scottrigby** [comment]
Also related to https://github.com/backstage/charts/pull/141 
Thanks @sabre1041 for pointing out

**Comment by joebowbeer** [comment]
@scottrigby also see: https://github.com/helm/helm-www/pull/1549

**Comment by anessi** [comment]
@scottrigby thanks for the review. I updated https://github.com/helm/helm-www/pull/1549 with the additional comment which makes totally sense.

**Comment by anessi** [comment]
Conflicts with main resolved.

**Comment by aldoborrero** [comment]
What else is missing for this PR to be merged?

**Comment by vepatel** [comment]
@aldoborrero actually waiting for this myself for a while, not sure what's taking this long

**Comment by anessi** [comment]
From what I see it's scheduled for the September milestone https://github.com/helm/helm/milestone/141
Not sure why it's not yet merged as I don't know the release process.

@sabre1041 @scottrigby : is there still something missing, or is this the usual process? Thanks for clarification.

**Comment by scottrigby** [comment]
> From what I see it's scheduled for the September milestone https://github.com/helm/helm/milestone/141 Not sure why it's not yet merged as I don't know the release process.
> 
> @sabre1041 @scottrigby : is there still something missing, or is this the usual process? Thanks for clarification.

@anessi yes it is scheduled to be released in September as part of 3.16.0, so we still have time to merge this. I will go ahead and merge now to avoid needing to resolve future conflicts with main.

Regarding the process question, we could have merged after the 2nd approval, I just had not until now. For a full description of the merging and release process, see https://github.com/helm/community/blob/main/helm-maintainers-onboarding-guide.md?plain=1#L150 from "Details on Merging PRs" down.

Note on the accompanying documentation PR: https://github.com/helm/helm-www/pull/1549 I have approved and added the `awaiting release` label, so that it can be merged once 3.16.0 is released.

**Comment by anessi** [comment]
@scottrigby : great! Thanks for merging and for the additional details about the process. Very helpful!

**Pull request merged**
